cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -774,6 +774,127 @@ TEST(Substrait, CallExtensionFunction) {
   }
 }
 
+TEST(Substrait, CallCast) {
+  ExtensionSet ext_set;
+  ConversionOptions conversion_options;
+
+  ASSERT_OK_AND_ASSIGN(auto buf,
+                       internal::SubstraitFromJSON("Expression", R"({
+  "selection": {
+      "directReference": {
+        "structField": {
```

Comment:
You use camelCase here but snake_case below (i.e. `direct_reference` and `struct_field`).  I was a bit surprised it works but it seems protobuf supports both.

However, let's pick one and be consistent.

Reply:
I am also surprised, didn't even know it could be supplied both ways - must have copied and pasted multiple bits from elsewhere.  Fixed now!